### PR TITLE
Move Remote features down to lower levels now that the RemoteHostClient is available from the Workspace layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @heejaechang @srivatsn @panopticoncentral 

I've moved FAR down.  I'm going to move NavigateTo down next.

**Comment by CyrusNajmabadi** [comment]
NavigateTo has been moved down as well.

File: src/Workspaces/Core/Portable/Remote/WellKnownServiceHubServices.cs [hunk 1]
```diff
@@ -16,13 +16,6 @@ internal static class WellKnownServiceHubServices
         public const string CodeAnalysisService_FindReferenceMethodsAsync = "FindReferenceMethodsAsync";
         public const string CodeAnalysisService_GetFullyQualifiedName = "GetFullyQualifiedName";
 
-        // NavigateTo
-        public const string CodeAnalysisService_SearchDocumentAsync = "SearchDocumentAsync";
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Remote/WellKnownServiceHubServices.cs]: thses are not needed anymore.  I just use 'nameof' to ensure that the names stay in sync.

**Comment by CyrusNajmabadi** [comment]
@heejaechang Can you take a look?

**Comment by heejaechang** [comment]
@CyrusNajmabadi I am looking

**Comment by CyrusNajmabadi** [comment]
Thanks :)

File: src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_FindReferences.cs [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
-using Microsoft.CodeAnalysis.Remote.Arguments;
+using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
     // root level service for all Roslyn services
-    internal partial class CodeAnalysisService
+    internal partial class CodeAnalysisService : IRemoteSymbolFinder
     {
         public async Task FindReferencesAsync(
```
Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_FindReferences.cs]: interesting.

Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_FindReferences.cs]: I guess this is what you talked about before

**Comment by heejaechang** [comment]
👍

**Pull request merged**